### DISCUSSION TITLE
comparing microphones - importance of quality microphone

### DISCUSSION THREAD

**Discussion by pddpauw** [mcguirepr89/BirdNET-Pi#1092] [Show and tell]
objective of this discussion, is to show the importance of good and qualitative microphones since they will result in more Processed detections.

In my set-up, I have 2 Pis:
Pi1: Rode NT USB microphone (approx 125 Eur), directly connected to a Pi4 (only running this birds analyzer software).  Using PulseAudio, with mic level at 92.
Pi2: cheap lavalier microphone (AliExpress, approx 20 Eur), connected to USB dongle, connected to a Pi5 (only running this birds analyzer software). Using PulseAudio, with mic level at 92.

Both have the same parameters (running V2.4, min conf level at 0.75 and sensitivity at 1.25).

Both microphones are 10cm away from each other, and recorded over the exact same time frame (3 days).  Weather was rainy and grey, so rather low number of detections, but it is the relative that counts.

Result of Pi1: 141 detections, 27 species detected (which seem to be correct)
![image](https://github.com/mcguirepr89/BirdNET-Pi/assets/148589775/39d0f8e7-18e5-44a5-b66c-0ded72791283)

Result of Pi2: 20 detections, 10 species detected (of which one is for sure not correct)
![image](https://github.com/mcguirepr89/BirdNET-Pi/assets/148589775/3aae2efa-5445-4df2-ab75-2dd96dc467a6)

So, the more expensive microphone has 7 times more processed detections and detected 3 time as many species.
(note that it might be that the cheaper microphone had more acoustic detections, but those didn't make it to to the Processed list due to -maybe- too low confidence level)

**Comment by tphakala** [comment]
Premise that "more expensive microphone is better" is just flawed, I have not yet found a better microphone than DIY ones based on PUI audio AOM5024 electret capsules, a capsule which costs ~3 USD.

**Reply by pddpauw** [reply]
this is also not what I wrote..., yet in my case; the more expensive one did better than the cheap one.
I wanted to show with this post that there are significant differences.

As a Christmasbreak project, I might in fact try to AOM5024 as well!

**Reply by stevedee** [reply]
Interesting results _pddpauw_, you have found a clear difference between your 2 microphones.

My tests were not clearcut; I thought it was going to be easy, but it wasn't.
I decided not to stress too much about it as the BirdNET-Pi system is really only taking a sample of local birds, and even then, they have to cooperate and sing or call, they have to be loud enough to overcome local noise and the system has to be reasonably confident with its identification. e.g. We get a lot of wood pigeons in our garden; they are very vocal in the summer, but practically mute in the winter! So from the system results, you might believe they are just a summer visitor.

For my base unit I'm using a relatively cheap electret mic capsule; Panasonic WM-61A. I suspect that if I could shield it from wind noise I'd get an even higher number of good detections/recordings.

For my portable unit I'm using a cheap, stubby USB mic because this is more convenient than the _mic + analogue/USB module_ combination used on my base unit.

More on my mic comparison testing: http://captainbodgit.blogspot.com/2023/11/birdnet-pi-portable-operation-mics-bird.html

**Reply by DMontgomery40** [reply]
I've had the EM272 capsules wired in parallel for stereo playback, and from everything I've read, that's about as good as it gets -- until I accidentally stumbled upon the HyperX QuadCast, completely by accident. I got it on a Black Friday sale for $79, normally about 40% more, and then you have to get the deadcat for it which is $15, but WOW.  SNR is 90db, and with a sensitivity of -36db, it is less sensitive than the EM272, but with a SNR 10db better than the EM272, the overall end result is still better, IMO, and for where I am. 

There are 4 polarity patterns, the Stereo option is fantastic, and an analog gain knob. With the analog gain turned all the way down, it is as quiet as the EM272; however, I turned it all the way up, because this installation was for my dad, and his hearing is bad. 

I can't figure out what capsules they are using, all the teardown videos on youtube are of gamers changing the led's and stuff like that (the LEDs do not go off, which would be highly annoying, but the deadcat takes care of that. 

Anyway, don't take my word for it, have a listen for yourself at either: 
https://app.birdweather.com/stations/1911
or
https://birdnet.dtmont.com/

Again, the DIY route with AOM5024s or the EM272s is still by far the best bang for your buck, but this is it for anyone who wants an amazing plug-and-play option. 

EDIT: 
Want to add some spectrograms of the QuadCast, this is set to 100% digital gain, with analog gain knob turned all the way to it's lowest setting. The confidence on a frequency that low is something I had not seen before, and this is over 4 different detections:  
<img width="1700" alt="Screenshot 2024-01-04 at 2 10 53 PM" src="https://github.com/mcguirepr89/BirdNET-Pi/assets/130489651/fbfbea76-551a-4144-ab17-d0ad9997f018">

On the higher frequencies, there have been some that I quite literally couldn't see, they were so dark/quiet, and all confidence rates above 75%; this one an example that's at least visible in the BirdNET UI, for most of the Juncos that are far away, I can only see them once I bring the spectrogram into Audacity. 

<img width="1700" alt="Screenshot 2024-01-04 at 2 24 02 PM" src="https://github.com/mcguirepr89/BirdNET-Pi/assets/130489651/cc53e34e-8c66-4973-9baa-10fe69cc4cee">

Like I mentioned earlier I recently cranked up the analog gain by A LOT, so my dad could hear the recordings better, and wow is that a powerful difference.  If you go to the station after 1/2/3023, you'll hear/see the loud stuff, prior to that, are the settings I've shown here.

**Comment by pddpauw** [comment]
ok, as promised, I have the results of the next test, and this came as a surprise!
@tphakala thanks for your advise after the previous post!

In my set-up, I have 2 Pis:
Pi1: Rode NT USB microphone (approx 125 Eur), directly connected to a Pi4 (only running this birds analyzer software). Using PulseAudio, with mic level at 92.
Pi2: DIY AOM microphone (2.70 Eur) soldered to audio jack (1.30 Eur) , connected to USB dongle (12 Eur), connected to a Pi5 (only running this birds analyzer software). Using PulseAudio, with mic level at 92. (I followed this instructions: https://github.com/mcguirepr89/BirdNET-Pi/discussions/39#discussioncomment-2180372; soldered including copper tape etc... with 1.5m wire)

Both have the same parameters (running V2.4, min conf level at 0.75 and sensitivity at 1.25).

Both microphones are 10cm away from each other, and recorded over the exact same time frame (20 hours). Weather was a combination of a bit of rain and occasional sunshine.

Result of Pi1: 471 detections, 26 species detected (could be correct, though doubting about Treecreeper, Certhia familiaris, might be mistaken with Short-toed Treecreeper, Certhia brachydactyla)
![image](https://github.com/mcguirepr89/BirdNET-Pi/assets/148589775/c1e05f25-4792-430b-a66b-b0bf68523a14)


Result of Pi2: 536 detections, 27 species detected (could be correct, though doubting about Treecreeper, Certhia familiaris, might be mistaken with Short-toed Treecreeper, Certhia brachydactyla)
![image](https://github.com/mcguirepr89/BirdNET-Pi/assets/148589775/2ad6426f-574a-492f-a9f9-3ad74e3a44a7)

The DIY AOM mic has 14% more detections.

Now, interestingly, let's take a species at the higher end of the frequency: Regulus regulus.
Pi1 made today 7 detections and median confidence level of 80%
![image](https://github.com/mcguirepr89/BirdNET-Pi/assets/148589775/72638d5b-c73e-49b5-9e9c-e4de8d7e3c3a)

Pi2 made today 13 detections and median confidence level of 75% (so it captured more, with probably a few that were harder to distinguish due to noice etc...)
![image](https://github.com/mcguirepr89/BirdNET-Pi/assets/148589775/7c6d3c5f-ce59-43c7-8432-58593ba65e72)

Let's take another one: Fringilla montifringilla
Pi1 made 1 detection with confidence level 86%
![image](https://github.com/mcguirepr89/BirdNET-Pi/assets/148589775/33f5f4be-0d89-47bc-b20b-b62dee736dc6)

Pi2 made 1 detection with confidence level 91%
![image](https://github.com/mcguirepr89/BirdNET-Pi/assets/148589775/c76efd5d-6a58-4ddb-aeb1-123b418c642e)

Listening to the audio recordings, I have the impression there is less noise on the AOM recordings and the sound is more clear.
However -and to give the complete picture- there are also examples in the other direction: in this case for the Anser albifrons:
Pi1 made 1 detection with confidence level 94%
![image](https://github.com/mcguirepr89/BirdNET-Pi/assets/148589775/035a1078-8366-4ff6-94ea-f11715bfe16b)

Pi2 made 1 detection with confidence level 87%
![image](https://github.com/mcguirepr89/BirdNET-Pi/assets/148589775/93e2b58e-ec43-469c-8bba-283bf5806d9d)


I'm not a sound or microphone expert, but I have not yet found the argument to pay a 100 Eur more for the Rode mic.
So, my field set ups will have teh AOM mic.

![WhatsApp Image 2024-01-04 at 09 24 22_b104b411](https://github.com/mcguirepr89/BirdNET-Pi/assets/148589775/9b45b2f8-c041-44ce-88aa-ad031e330376)
![WhatsApp Image 2024-01-04 at 19 40 26_bec812a2](https://github.com/mcguirepr89/BirdNET-Pi/assets/148589775/a49478c4-ca73-4750-8220-3158e986f017)

**Reply by tphakala** [reply]
If you are looking for cheap sound card to go with AOM5024 capsules search for cm108 sound card in ebay and aliexpress, I recently got 10 pcs for less than 30 euros. cm108 cards provide 4.5v plugin power which is higher than many brand name usb sound cards, AOM5024 bias voltage range is from 1 to 10v and higher voltage increases capsule sensitivity.

![41glN8VU7xL](https://github.com/mcguirepr89/BirdNET-Pi/assets/7030001/e9f2714a-b332-44e9-896e-eca6062d8b79)

I use Focusrite scarlett audio interfaces at my home station with simple p48 wiring to AOM5024, but I plan to install some remote stations to monitor owl boxes and for this I am going to use cm108 sound cards.

**Reply by Astrobirder** [reply]
@pddpauw THANK YOU for doing this experiment, excellent work and perfectly timed for me!  I'm about to update my Birdnet setup in preparation for semi-permanent deployment (mounting in a weatherproof box).  One question--what USB sound card were you using for your tests?  I've tried a couple (I'll have to go look to see which ones, but I know one is the UGREEN that was recommended).

**Reply by pddpauw** [reply]
For now, I've used the [Adafruit](https://www.digikey.be/nl/products/detail/adafruit-industries-llc/1475/7244940?utm_adgroup=&utm_source=google&utm_medium=cpc&utm_campaign=PMax%20Shopping_Product_Low%20ROAS&utm_term=&productid=7244940&utm_content=&utm_id=go_cmp-20168770009_adg-_ad-__dev-c_ext-_prd-7244940_sig-CjwKCAiA1-6sBhAoEiwArqlGPs-DWpTFr_TIRQRK-X5-gqjOTMqAaOU-7ptdIWQJM9xFr2eL1FI2hxoCVpUQAvD_BwE&gad_source=1&gclid=CjwKCAiA1-6sBhAoEiwArqlGPs-DWpTFr_TIRQRK-X5-gqjOTMqAaOU-7ptdIWQJM9xFr2eL1FI2hxoCVpUQAvD_BwE) one.
Maybe that one day I will do a check Adafruit vs Green (I have both), but I need to find the time to do so...

**Reply by stevedee** [reply]
I'd be interested to know what is inside the UGREEN module.
I have 3 PiHut/Adafruit white modules which I'd assumed were all CM108 based, but one actually uses a Texas Instruments PCM2902 chip.
I'm not sure that these retailers know what they are selling.

In Web Terminal or a Linux terminal on BirdNET-Pi just run: **_lsusb_**
...and post the output.

**Reply by tphakala** [reply]
I have UGREEN usb sound card, somewhere, when I was comparing plug-in voltages of several sound cards UGREEN had  one of lowest voltages I'v measured, based on that some electret capsules are less sensitive with that card than on other cards. Those cheap cm108 cards I referred on other post are far better.

**Reply by DMontgomery40** [reply]
@stevedee UGREEN probably has more than one vendor but I've had two UGREEN 3.5mm TRRS to USB-A ADC/DAC and the output for both is:  "KT Audio"

edit:  OOPS!!  that was how it was recognized by alsa, lsusb output: 

```
dmontg@raspberrypi:~ $ lsusb
Bus 002 Device 001: ID 1d6b:0003 Linux Foundation 3.0 root hub
Bus 001 Device 002: ID 2109:3431 VIA Labs, Inc. Hub
Bus 001 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
```

**Reply by stevedee** [reply]
Thanks DM. I was hoping to see something like this...
![BirdnetPi_lsusb](https://github.com/mcguirepr89/BirdNET-Pi/assets/1870807/95350677-60ce-45aa-a45d-30c9770a9d7b)
...as a search for KT Audio finds too many companies. But it could be this one: http://www.ktmicro.com/?usb_73/179.html

For me, the most important parameters are signal-to-noise ratio and low distortion. If the S/N ratio is high enough, then small variations in sensitivity can be overcome by adjusting the gain. I sometimes get clipping if the wind is strong enough, but I can't face climbing a ladder to fit some kind of fluffy thing to the mic on my roof!

My TI based USB module is mono but fitted with a stereo chip. It has a relatively low mic bias voltage of 1.5Volts but seems to perform much like my two CM108 based modules with bias of 2Volts. I also think that any attempt to record with more than one mic (stereo) could be detrimental to analysis, as it may result in phase distortion.

One problem with analogue audio-to-USB modules on a RaspberryPi using a recommended mains power supply is common mode noise. This is due to the Pi 0V/common floating around above ground potential. You can check this for yourself by temporarily using a powerbank. The most obvious difference is usually the sound of mains hum (50Hz UK or 60Hz North America).

On my 'base' system I've connected Pi 0V to earth (ground) which considerably reduces noise. On my 'portable' system I use a power bank and a USB mic, so not a problem. The best solution for a mains powered system would probably be an 'old school' power supply fitted with a proper isolation transformer!

**Reply by DMontgomery40** [reply]
![image](https://github.com/mcguirepr89/BirdNET-Pi/assets/130489651/8df8d0a6-3b82-4b32-8283-58951920dc73)

Okay there we go, UGREEN needed a Mic plugged in to show up. This is my indoor Pi, my BirdNET-Pi is on the roof like yours.  The reason I'm not using this, or the EM272/Clippy mic anymore, is because of the HyperX QuadCast, and if amazing SNR interests you, the HyperX QuadCast is 90db and [it's amazing ](https://github.com/mcguirepr89/BirdNET-Pi/discussions/1092#discussioncomment-8056571:~:text=spectrograms%20of%20the%20QuadCast), I'm currently running it side-by-side with the EM272 on a Movo soundcard for a longterm comparison.  I don't know much about audio at all, I just know this QuadCast thing has an ink-black spectrogram and it hears more birds with higher accuracy; here's today (but a longer term comparison will show the best results):

![image](https://github.com/mcguirepr89/BirdNET-Pi/assets/130489651/c81aa278-b27d-4443-8aa4-f8abedbf92af)

**Reply by DD4WH** [reply]
Thanks for the comparisons!

In the case of the HyperX Quadcast, I think the SNR you report is not the SNR of the mic, but of its headphone amp. Have a look here: https://support.hyperx.com/Microphones/Quadcast/2474801931/QuadCast-Specifications.htm

So, I am not convinced about the mentioned mic having less self noise than the EM272. Be careful to accurately adjust the levels of the mic before comparing spectrograms. The EM272 is much more sensitive, so naturally the spectrogram is filled with more color ... Comparing self noise of mics is complicated.

Have fun with the BirdNET-Pi!

**Reply by stevedee** [reply]
Note: When comparing microphones, you could use a smart phone with a signal generator app to set reasonably consistent gain/sensitivity levels for each system. Microphone sensitivity may vary, even when comparing individual microphones of the same type.

**Comment by Astrobirder** [comment]
@pddpauw Thanks for this timely info!  I'm about to update my Birdnet-Pi setup. A quick question regarding test setups, can you tell me what USB sound card you were using in both test 1 and 2?

**Comment by stevedee** [comment]
I carried out some microphone testing to compare sensitivity and to look at noise: http://captainbodgit.blogspot.com/2024/01/birdnet-pi-more-thoughts-on-microphones.html
I think sensitivity testing is fairly straight forward, if you want to compare a few microphones.

**Comment by DMontgomery40** [comment]
@DD4WH this is such a crazy rabbit hole you've sent me down lol..  I had 4 solid weeks of data comparing the QuadCast to the em272, and long story, lost the em272 data, redoing it now. I want more than a day or two, I want a lot of data.  But, the outcome was really interesting...

For two weeks the quadcast was getting close to 50% more detections. This didn't seem right. I switched from a $15 movo soundcard to a $90 Rode AI soundcard. My wife wanted to murder me, but it's for science! 

The difference between: 
em272 + movo 
--and-
em272 + rode 
was bigger than the difference between the rather poor powerDewise plus ugreen.  It is wild how impactful that soundcard was. 

But...  the really crazy part is, it is still head-to-head with the QuadCast.  I think you're right about my misread on the 90 SNR being for the headphone output; however, it is still performing at the same level of the em272. 

I feel like that guy who was arguing with you for a solid year about how the Rode shotgun mic was better than anything else no matter how many datasheets were thrown at him, and the fact that he didn't have a birdnet setup at all...  I don't want it to come across like I'm arguing that the QuadCast is just as good, I think I'm just so deep into learning about this stuff, I have to figure out WHY I'm seeing these results. 

Every other variable in this experiment has been accounted for, and again, I'm going to come back with some "long" term data here.

The only thing I can think of is that the QuadCast has three capsules. I don't know enough about audio to know if that's a valid reason at all, I would think more does not equal better, but maybe if you're just just performing at 50% of the em272 and you have 150% more capsules (facing 3 different directions), you can compete? 

Anyway I don't want to speculate further, I'm a big fan of data and would like to just use that as opposed to my own uninformed narrative. 

@Astrobirder if you have $90 burning a hole in your pocket and you haven't gotten your sound card yet -- look no further (I've tried 5):  https://rode.com/en-us/interfaces-and-mixers/ai-series/ai-micro

**Reply by DD4WH** [reply]
I am sorry for that! It was not my intention to send you down a rabbit hole :-).
The tests you are performing are quite complex. I think you are comparing many things at the same time and that makes it very difficult. Your aim is to compare the mics. The easiest thing would be to compare the mics with a test tone and plot the result as a spectrogram (without BirdNET-Pi and without anything else). Then you would change the sound card and measure that effect, And only in the very end you would test the performance with BirdNET.

The first thing I would do to be sure that my tests are really testing the things I want to test, would be:
- to test two BirdNET-Pis (same RPi model) with the **same mic**, the **same mic wind screen** and the **same sound** card and the **same power supply** side-by-side (only a few centimetres apart) and have a look at the results. If that test already reveals differences, then all your complex test result differences you already noticed could be traced back to the different position of the BirdNET-Pi mics (even if its only a few cm). And do not forget (as has already been mentioned by others) that even the same mic model can have a little different sensitivity and thus lead to different no. of detections. 

Good luck and have fun with your comparisons!

**Reply by alexbelgium** [reply]
Hi, very interesting tests! I've invested in both to also test (the crazy things that we do  in the name of science :))

In the end, which system between EM272 + Rode AI vs Hyperx Quadcast did you keep? Also, did your quadcast survive outdoor for long? I am thinking about protecting it under a shed roof but there will be plenty of atmospheric humidity when raining or during the night. Thanks!

Here is my experience with both :
- Test system
    - Microphone 
        - Hardware : HyperX Quadcast with deadcat ; and UGREEN aux/usb converter with chip KT0210 + Clippy EM272 smart TTRS with deadcat. Both systems have sensibilities described up to 20khz
        - Alsamixer setting : 67% for HyperX (gain +3db) ; 100% for Clippy EM272 (gain +0db)
        - RPI3b+ connected to both microphones via USB. Both outputs are sent through a RTSP stream using the formula : `ffmpeg -nostdin -f alsa -acodec pcm_s16le -ac 1 -ar 48000 -i hw:1,0 -f rtsp -acodec pcm_s16be rtsp://localhost:8554/birdmic -rtsp_transport tcp`
        - Location : Both microphones are placed against the same wall of a shed, with a distance of around 0.5m between both mics (facing same direction)
    - Analyzer
         - Rpi4b 8go ram with HomeAssistant installed, with two BirdNET-Pi addons running in parallel (exact same setting)
         - Each BirdNET-Pi analyses a different RTSP stream from the microphones
         - Settings : recording length 15s ; overlap 0,5s ; confidence 0,7 ; sensitivity 1,25
    - Results
        - Spectrograms are very comparable : same noise, very comparable outputs (here with the same sound recorded on both)
       <img src="https://github.com/mcguirepr89/BirdNET-Pi/assets/44178713/a0330901-ef41-4bc1-81a9-54c29e494b96.png)" width="300">

        - Results is that Clippy with EM272 is a bit more more sensitive ; although it seems both mics have preferences in species detected (and even in real time listening, the sound is a bit different)
        <img src="https://github.com/mcguirepr89/BirdNET-Pi/assets/44178713/9de20cf8-4e98-4097-9eae-a56ab44a595e.png)" width="300">
